service/models.py

```diff
@@ -51,6 +51,15 @@ def create(self):
         db.session.add(self)
         db.session.commit()
 
+    def update(self):
+        """
+        update a recommendation type to the database
+        """
+        if not self.relationship:
+            raise DataValidationError("Update called with empty relationship")
```

Comment:
What I mean below is that I think you could remove these two lines

Reply:
Right, it makes sense. But I think I can leave here for double check.

Reply:
Then you want to test it in test_model.py because route will never hit here.

Reply:
already commit new version that test it in  test_model.py . Thanks